extmod/modnetwork.c

```diff
@@ -122,7 +122,7 @@ STATIC mp_obj_t network_hostname(size_t n_args, const mp_obj_t *args) {
     } else {
         size_t len;
         const char *str = mp_obj_str_get_data(args[0], &len);
-        if (len >= MICROPY_PY_NETWORK_HOSTNAME_MAX_LEN) {
+        if (len > MICROPY_PY_NETWORK_HOSTNAME_MAX_LEN) {
```

Comment:
Would it be worth putting this block of code into a (static inline?) function, so it doesn't need to be repeated 4 times?  Like in the previous PR.

Reply:
Done, in a second commit.

Rather than making a static inline, which would result in two copies per port, I just made the `if.config` version use the `network.hostname` implementation directly, which might actually be a small code size saving.